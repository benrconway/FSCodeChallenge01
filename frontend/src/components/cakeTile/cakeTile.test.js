import React from "react";
import { render } from "@testing-library/react";
import CakeTile from "./cakeTile";
import { BrowserRouter } from "react-router-dom";

const testCake = {
  id: 1,
  name: "__TEST_CAKE__",
  comment: "__TEST_COMMENT__",
  imageUrl: "__TEST_URL__",
  yumFactor: 5,
};

test("renders CakeTile screen", () => {
  const { getByText } = render(
    <BrowserRouter>
      <CakeTile cake={testCake} fullDescription={false} />
    </BrowserRouter>
  );
  const linkElement = getByText(/More/i);
  expect(linkElement).toMatchSnapshot();
});
