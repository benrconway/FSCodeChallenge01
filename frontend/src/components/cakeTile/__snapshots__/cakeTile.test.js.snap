// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CakeTile screen 1`] = `
<a
  href="/cakes/1"
>
  More details
</a>
`;
